happi/containers.py

```diff
@@ -1,330 +1,27 @@
-    system by default.
-
-    We're simplifying here, assuming the LODCM is aligned and that the states
-    are accurate. We'll probably only look at the H1N state for lightpath
-    purposes, but it's good to collect all the available information.
-
-    Attributes
-    ----------
-    prefix : str
-        The base PV should be the state PV associated with the h1n or h2n
-        state, depending on which crystal we're referring to e.g.
-        "XPP:LODCM:H1N".
-
-    mono_line : str
-        Name of the mono line
-    """
-    device_class = copy(Device.device_class)
-    device_class.default = 'pcdsdevices.device_types.LODCM'
-    mono_line = EntryInfo("Name of the MONO beamline",
-                          enforce=str, optional=False)
-    kwargs = deepcopy(Device.kwargs)
-    kwargs.default.update({'mono_line': '{{mono_line}}',
-                           'main_line': '{{beamline}}'})
-
-
-class MovableStand(Device):
-    """
-    This class stores information about stands that move, like XPP's hand-crank
-    that moves SB2 and SB3 from the PINK to XPP lines and back. There is no
-    need to instantiate one of these for static stands.
-
-    Attributes
-    ----------
-    prefix : str
-        If there is a single PV with the stand's location, this should be the
-        base PV. In general, these devices will actually have multiple PVs
-        with binary outputs that have yes/no on the stand being in each
-        position. In these cases we pick the common prefix of these PVs.
-
-    stand : list
-        List of stands affected by table movement.
-    """
-    stand = copy(Device.stand)
-    stand.enforce = list
-    system = copy(Device.system)
-    system.default = 'changeover'
-
-
-class Motor(Device):
-    """
-    A Generic EpicsMotor
-    """
-    device_class = copy(Device.device_class)
-    device_class.default = 'pcdsdevices.device_types.Motor'
-    system = copy(Device.system)
-    system.default = 'motion'
-
-
-class AreaDetector(Device):
-    """
-    A Generic EpicsCamera
-    """
-    device_class = copy(Device.device_class)
-    device_class.default = 'pcdsdevices.device_types.PCDSDetector'
-    system = copy(Device.system)
-    system.default = 'camera'
-
-
-class Acromag(Device):
-    """
-    A Generic class for Acromag
-    """
-    device_class = copy(Device.device_class)
-    device_class.default = 'pcdsdevices.device_types.Acromag'
-    system = copy(Device.system)
-    system.default = 'acromag'
-
-
-class Trigger(Device):
-    """
-    A Generic class for Controls Triggers
-    """
-    device_class = copy(Device.device_class)
-    device_class.default = 'pcdsdevices.device_types.Trigger'
-    system = copy(Device.system)
-    system.default = 'timing'
+import logging
+import inspect
+import entrypoints
+
+from .item import HappiItem
+
+logger = logging.getLogger(__name__)
+
+_entries = entrypoints.get_group_all('happi.containers')
+
+registry = {}
```

Comment:
Let's move the `registry` generation code inside a function, otherwise things like `name`, `entry`, `obj`, etc may leak such that `from happi.containers import entry` would work.

Reply:
Will address that.